test/unit/grpcServer/handlers/core/getLastStateTransitionHashHandlerFactory.js

```diff
@@ -42,39 +42,17 @@ describe('getLastStateTransitionHashHandlerFactory', () => {
     );
   });
 
-  it('should throw an error if userId is not a buffer', function it(done) {
-    const userId = 'someStringId';
-
-    const call = new GrpcCallMock(this.sinon, {
-      userId,
-    });
-
-    const callback = (e, v) => {
-      try {
-        expect(v).to.equal(null);
-        expect(e).to.be.an.instanceOf(InvalidArgumentError);
-        expect(e.getMessage()).to.equal('Invalid argument: userId is not a buffer');
-
-        done();
-      } catch (error) {
-        done(error);
-      }
-    };
-
-    coreAPIMock.getUser.resolves(undefined);
-
-    getLastStateTransitionHashHandler(call, callback);
-  });
-
   it('should throw an error if userId is not of correct length', function it(done) {
     const userId = Buffer.from('someStringId');
 
     const call = new GrpcCallMock(this.sinon, {
       userId,
     });
 
-    const callback = (e, v) => {
+    const callback = sinon.spy((e, v) => {
       try {
+        expect(callback).to.have.been.calledOnce();
```

Comment:
We need to check that callback called outside callback, 'coz if callback will not be called then your checks will not be run.

Reply:
@shumkov I'm not sure how to do that since we have a callback here. I need to call `done` somewhere, but if I call it inside the spy, then the checks that go afterwards are not executed. And if I call `done` in the end of `it` block there is a chance that `callback` is not executed.

Reply:
You need to call `done` after your checks in callback I think. So in case if callback will not be called you will get timeout error.

Reply:
And for sure you don't need spy in this case.